pkgs/tools/security/sbomnix/default.nix

```diff
@@ -1,55 +1,271 @@
-{ lib
-, fetchFromGitHub
-, coreutils
-, curl
-, gnugrep
-, gnused
-, gzip
-, nix
-, python
+{
+  lib,
+  fetchFromGitHub,
+  grype,
+  nix,
+  python,
+  vulnix,
   # python libs
-, colorlog
-, graphviz
-, numpy
-, packageurl-python
-, pandas
-, requests
-, reuse
-, tabulate
+  beautifulsoup4,
+  binaryornot,
+  boolean-py,
+  colorlog,
+  dataproperty,
+  debian,
+  graphviz,
+  jinja2,
+  license-expression,
+  matplotlib,
+  mbstrdecoder,
+  numpy,
+  networkx,
+  packageurl-python,
+  packaging,
+  pandas,
+  pathvalidate,
+  poetry-core,
+  pygraphviz,
+  requests,
+  requests-cache,
+  reuse,
+  tabledata,
+  tabulate,
+  typepy,
+  typing-extensions
 }:
-
 python.pkgs.buildPythonApplication rec {
   pname = "sbomnix";
-  version = "1.4.5";
+  version = "1.6.1";
 
   src = fetchFromGitHub {
     owner = "tiiuae";
-    repo = pname;
+    repo = "sbomnix";
     rev = "refs/tags/v${version}";
-    hash = "sha256-RxDFxVGivVBw2uhtzf231Q2HHTBFKSqGrknr2Es/ygM=";
+    hash = "sha256-dECeNsJgsN/LTXas5HchoJJHpY73xWuTK+9A0budGOU=";
   };
 
-  makeWrapperArgs = [
-    "--prefix PATH : ${lib.makeBinPath [ coreutils curl gnugrep gnused gzip graphviz nix ]}"
-  ];
+  postInstall = ''
+
+    wrapProgram $out/bin/sbomnix \
+        --prefix PATH : ${lib.makeBinPath [nix graphviz]}
+
+    wrapProgram $out/bin/nixgraph \
+        --prefix PATH : ${lib.makeBinPath [nix graphviz]}
+
+    wrapProgram $out/bin/vulnxscan \
+        --prefix PATH : ${lib.makeBinPath [grype nix vulnix-patched]}
+
+    wrapProgram $out/bin/nix_outdated \
+        --prefix PATH : ${lib.makeBinPath [nix-visualize]}
+
+    wrapProgram $out/bin/provenance \
+        --prefix PATH : ${lib.makeBinPath [nix]}
+
+  '';
+
+  # We need to patch vulnix to add support for runtime-only scan
+  # ('-C' command-line option) which is currently not available in
+  # released version of vulnix.
+  # Pending: https://github.com/nix-community/vulnix/pull/80.
+  vulnix-patched = vulnix.overrideAttrs (
+    old: rec {
+      patches = [
```

Comment:
Maybe use `patches = (old.patches or []) ++ [` so that if vulnix had some patches, they are kept?

Reply:
Thanks, this is fixed in [6a9d99c](https://github.com/NixOS/nixpkgs/commit/6a9d99cca1b6ba639e537be7e0a33447a9f3bf27).